### DISCUSSION TITLE
Multiple configuration sources by convention

### DISCUSSION THREAD

**Discussion by gavar** [directus/directus#18839] [Feature Requests]
### Summary

It would be great to implement a way to search for multiple configuration candidates when initializing cli. Currently, `directus` cli requires to provide a `CONFIG_PATH` environment value to override the way of loading configurations. This leads to a problem where you need to set  `CONFIG_PATH` environment variable before every CLI invocation.

### Motivation

Make a developer-friendly configuration approach supporting various configuration source that does not need to alter environment variables so that the local development become easier.

### Detailed Design

A solution might be to introduce an automatic way of searching the first valid configuration file which might be `.ts`, `.js`, `.yaml`, `.json` or `.env`.  A very common tool for that is https://www.npmjs.com/package/cosmiconfig which is already a peer dependency in a project.

### Must Have:
- a configuration file naming convention to search for configuration sources
- find first available configuration source from list of defaults source and stick to that

### Should Have:
- support of most popular configuration sources like  `.ts`, `.js`, `.yaml`, `.json` or `.env`

### Won't Have:
- hardcoded default configuration source of `.env`

### Drawbacks

There are no drawbacks of that solution.

### Alternatives

An alternative solution might be to use https://www.npmjs.com/package/rc-config-loader but it's way less popular. 

### Adoption Strategy

The proposal does not introduce breaking changes and does not require any migration.

For projects providing CONFIG_PATH explicitly the configuration stays the same.

Projects that do not use CONFIG_PATH should be informed which configuration files has been chosen to run the setup.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
